modbuscommander/integrationpluginmodbuscommander.cpp

```diff
@@ -63,94 +64,45 @@ void IntegrationPluginModbusCommander::init()
-
-    } else if ((thing->thingClassId() == coilThingClassId)
-               || (thing->thingClassId() == discreteInputThingClassId)
-               || (thing->thingClassId() == holdingRegisterThingClassId)
-               || (thing->thingClassId() == inputRegisterThingClassId)) {
-        info->finish(Thing::ThingErrorNoError);
-
-    } else {
-        Q_ASSERT_X(false, "setupThing", QString("Unhandled thingClassId: %1").arg(thing->thingClassId().toString()).toUtf8());
-    }
-}
-
 void IntegrationPluginModbusCommander::discoverThings(ThingDiscoveryInfo *info)
 {
     ThingClassId thingClassId = info->thingClassId();
 
     if (thingClassId == modbusRTUClientThingClassId) {
         Q_FOREACH(QSerialPortInfo port, QSerialPortInfo::availablePorts()) {
-            //Serial port is not yet used, create now a new one
-            qCDebug(dcModbusCommander()) << "Found serial port:" << port.systemLocation();
-            QString description = port.manufacturer() + " " + port.description();
-            ThingDescriptor thingDescriptor(thingClassId, port.portName(), description);
+            qCDebug(dcModbusCommander()) << "Found serial port:" << port.systemLocation() << "manufacturer" << port.manufacturer() << "description" << port.description() << "serial number" << port.serialNumber();
+            if (port.isBusy()) {
+                qCDebug(dcModbusCommander()) << "Serial port ist busy, skipping.";
+                continue;
+            }
+            QString manufacturer = port.manufacturer();
+            if (manufacturer.isEmpty()) {
+                manufacturer = "unknown";
+            }
+            QString description = port.description()+" Manufacturer: "+port.manufacturer();
+            ThingDescriptor thingDescriptor(thingClassId, "Modbus RTU interface", description);
             ParamList parameters;
             QString serialPort = port.systemLocation();
-            foreach (Thing *existingThing, myThings()) {
-                if (existingThing->paramValue(modbusRTUClientThingSerialPortParamTypeId).toString() == serialPort) {
-                    thingDescriptor.setThingId(existingThing->id());
-                    break;
-                }
+            QString serialnumber = port.serialNumber();
+            if (serialnumber.isEmpty()) {
+                serialnumber = port.manufacturer()+QString::number(port.productIdentifier(), 16);
+
+            }
+            qCDebug(dcModbusCommander()) << "    - Serial number" << serialnumber;
+            Q_FOREACH (Thing *exisingThing, myThings().filterByParam(modbusRTUClientThingClassId)) {
+                thingDescriptor.setThingId(exisingThing->id());
+                // Rediscovery is broken because of an missing unique device id
+                // This is a workaround and doesnt work if multiple uart converters are attached.
+                // ThingDiscoveryInfo may be extended to distinquish between discovery and rediscovery
```

Comment:
how would that solve this issue?

Reply:
In case of an rediscovery the thingId would be added to every available and not busy serial port, for that the thingId is required on discovery. This thing class is special, not only do I have an issue with the missing ID of usb-uart convers  but also the user may want to replace the serial port with another one (and move all the child things to the new serial port).

Reply:
got it... makes sense... Will incorporate this thought when touching this stuff next time.